[flaky tests] TestReshardPartialBatch

See https://github.com/prometheus/prometheus/actions/runs/21958882084/job/63430579331?pr=18071

```
goroutine profile: total 189
72 @ 0x4bdf2e 0x4992f7 0x1c155b6 0x4c6a41
#	0x1c155b5	github.com/prometheus/prometheus/storage/remote.(*shards).runShard+0x935	/__w/prometheus/prometheus/storage/remote/queue_manager.go:1610

65 @ 0x4c6a41

5 @ 0x4bdf2e 0x4992f7 0x187c665 0x4c6a41
#	0x187c664	github.com/prometheus/prometheus/tsdb/wlog.(*Watcher).loop+0x484	/__w/prometheus/prometheus/tsdb/wlog/watcher.go:272

5 @ 0x4bdf2e 0x4992f7 0x1c1068f 0x4c6a41
#	0x1c1068e	github.com/prometheus/prometheus/storage/remote.(*QueueManager).updateShardsLoop+0x1ee	/__w/prometheus/prometheus/storage/remote/queue_manager.go:1104

4 @ 0x4bdf2e 0x4992f7 0x1c12058 0x4c6a41
#	0x1c12057	github.com/prometheus/prometheus/storage/remote.(*QueueManager).reshardLoop+0x1d7	/__w/prometheus/prometheus/storage/remote/queue_manager.go:1231

3 @ 0x4bdf2e 0x47d837 0x4bd0e5 0x530191 0x531d73 0x531d49 0x83c90b 0x85714d 0xa40148 0x5f43a4 0x5f4645 0xa49e5b 0x4c6a41
#	0x4bd0e4	internal/poll.runtime_pollWait+0x84		/usr/local/go/src/runtime/netpoll.go:351
#	0x530190	internal/poll.(*pollDesc).wait+0xb0		/usr/local/go/src/internal/poll/fd_poll_runtime.go:84
#	0x531d72	internal/poll.(*pollDesc).waitRead+0x452	/usr/local/go/src/internal/poll/fd_poll_runtime.go:89
#	0x531d48	internal/poll.(*FD).Read+0x428			/usr/local/go/src/internal/poll/fd_unix.go:165
#	0x83c90a	net.(*netFD).Read+0x4a				/usr/local/go/src/net/fd_posix.go:68
#	0x85714c	net.(*conn).Read+0xac				/usr/local/go/src/net/net.go:196
#	0xa40147	net/http.(*connReader).Read+0x287		/usr/local/go/src/net/http/server.go:812
#	0x5f43a3	bufio.(*Reader).fill+0x2a3			/usr/local/go/src/bufio/bufio.go:113
#	0x5f4644	bufio.(*Reader).Peek+0xc4			/usr/local/go/src/bufio/bufio.go:152
#	0xa49e5a	net/http.(*conn).serve+0xffa			/usr/local/go/src/net/http/server.go:2145

3 @ 0x4bdf2e 0x47d837 0x4bd0e5 0x530191 0x531d73 0x531d49 0x83c90b 0x85714d 0xa77b75 0x5f43a4 0x5f4645 0xa78aea 0x4c6a41
#	0x4bd0e4	internal/poll.runtime_pollWait+0x84		/usr/local/go/src/runtime/netpoll.go:351
#	0x530190	internal/poll.(*pollDesc).wait+0xb0		/usr/local/go/src/internal/poll/fd_poll_runtime.go:84
#	0x531d72	internal/poll.(*pollDesc).waitRead+0x452	/usr/local/go/src/internal/poll/fd_poll_runtime.go:89
#	0x531d48	internal/poll.(*FD).Read+0x428			/usr/local/go/src/internal/poll/fd_unix.go:165
#	0x83c90a	net.(*netFD).Read+0x4a				/usr/local/go/src/net/fd_posix.go:68
#	0x85714c	net.(*conn).Read+0xac				/usr/local/go/src/net/net.go:196
#	0xa77b74	net/http.(*persistConn).Read+0xf4		/usr/local/go/src/net/http/transport.go:2125
#	0x5f43a3	bufio.(*Reader).fill+0x2a3			/usr/local/go/src/bufio/bufio.go:113
#	0x5f4644	bufio.(*Reader).Peek+0xc4			/usr/local/go/src/bufio/bufio.go:152
#	0xa78ae9	net/http.(*persistConn).readLoop+0x329		/usr/local/go/src/net/http/transport.go:2278

3 @ 0x4bdf2e 0x47d837 0x4bd0e5 0x530191 0x5380ca 0x5380a0 0x83fcc5 0x866cae 0x865065 0xa525f0 0x1a133f3 0x4c6a41
#	0x4bd0e4	internal/poll.runtime_pollWait+0x84		/usr/local/go/src/runtime/netpoll.go:351
#	0x530190	internal/poll.(*pollDesc).wait+0xb0		/usr/local/go/src/internal/poll/fd_poll_runtime.go:84
#	0x5380c9	internal/poll.(*pollDesc).waitRead+0x4e9	/usr/local/go/src/internal/poll/fd_poll_runtime.go:89
#	0x53809f	internal/poll.(*FD).Accept+0x4bf		/usr/local/go/src/internal/poll/fd_unix.go:613
#	0x83fcc4	net.(*netFD).accept+0x44			/usr/local/go/src/net/fd_unix.go:161
#	0x866cad	net.(*TCPListener).accept+0x4d			/usr/local/go/src/net/tcpsock_posix.go:159
#	0x865064	net.(*TCPListener).Accept+0x64			/usr/local/go/src/net/tcpsock.go:380
#	0xa525ef	net/http.(*Server).Serve+0x60f			/usr/local/go/src/net/http/server.go:3463
#	0x1a133f2	net/http/httptest.(*Server).goServe.func1+0xb2	/usr/local/go/src/net/http/httptest/server.go:311

3 @ 0x4bdf2e 0x4992f7 0x1c03dff 0x4c6a41
#	0x1c03dfe	github.com/prometheus/prometheus/storage/remote.(*MetadataWatcher).loop+0x21e	/__w/prometheus/prometheus/storage/remote/metadata_watcher.go:119

3 @ 0x4bdf2e 0x4992f7 0xa7b6d6 0x4c6a41
#	0xa7b6d5	net/http.(*persistConn).writeLoop+0x1b5	/usr/local/go/src/net/http/transport.go:2600

2 @ 0x4bdf2e 0x4992f7 0x1c28fa9 0x4c6a41
#	0x1c28fa8	github.com/prometheus/prometheus/storage/remote.(*WriteStorage).run+0x188	/__w/prometheus/prometheus/storage/remote/write.go:118

1 @ 0x1c14b5c 0x1c13eb4 0x1c157a8 0x4c6a41
#	0x1c14b5b	github.com/prometheus/prometheus/storage/remote.(*queue).newBatch+0x25b		/__w/prometheus/prometheus/storage/remote/queue_manager.go:1537
#	0x1c13eb3	github.com/prometheus/prometheus/storage/remote.(*queue).Batch+0x133		/__w/prometheus/prometheus/storage/remote/queue_manager.go:1473
#	0x1c157a7	github.com/prometheus/prometheus/storage/remote.(*shards).runShard+0xb27	/__w/prometheus/prometheus/storage/remote/queue_manager.go:1642

1 @ 0x457235 0x1c151da 0x4c6a41
#	0x1c151d9	github.com/prometheus/prometheus/storage/remote.(*shards).runShard+0x559	/__w/prometheus/prometheus/storage/remote/queue_manager.go:1574

1 @ 0x479e11 0x4bccdd 0x7490ca 0x748e86 0x7444f4 0x1c63310 0x62a53d 0x4c6a41
#	0x7490c9	runtime/pprof.writeRuntimeProfile+0xc9							/usr/local/go/src/runtime/pprof/pprof.go:788
#	0x748e85	runtime/pprof.writeGoroutine+0x85							/usr/local/go/src/runtime/pprof/pprof.go:747
#	0x7444f3	runtime/pprof.(*Profile).WriteTo+0x1b3							/usr/local/go/src/runtime/pprof/pprof.go:371
#	0x1c6330f	github.com/prometheus/prometheus/storage/remote.TestReshardPartialBatch.func1+0x52f	/__w/prometheus/prometheus/storage/remote/queue_manager_test.go:657
#	0x62a53c	testing.tRunner+0x21c									/usr/local/go/src/testing/testing.go:1934
1 @ 0x4bdf2e 0x44e685 0x44e232 0x62c15e 0x1c614ca 0x62a53d 0x4c6a41
#	0x62c15d	testing.(*T).Run+0x9fd							/usr/local/go/src/testing/testing.go:2005
#	0x1c614c9	github.com/prometheus/prometheus/storage/remote.TestReshard+0xa9	/__w/prometheus/prometheus/storage/remote/queue_manager_test.go:549
#	0x62a53c	testing.tRunner+0x21c							/usr/local/go/src/testing/testing.go:1934

1 @ 0x4bdf2e 0x44e685 0x44e232 0x62c15e 0x1c6204a 0x62a53d 0x4c6a41
#	0x62c15d	testing.(*T).Run+0x9fd								/usr/local/go/src/testing/testing.go:2005
#	0x1c62049	github.com/prometheus/prometheus/storage/remote.TestReshardRaceWithStop+0xa9	/__w/prometheus/prometheus/storage/remote/queue_manager_test.go:588
#	0x62a53c	testing.tRunner+0x21c								/usr/local/go/src/testing/testing.go:1934

1 @ 0x4bdf2e 0x44e685 0x44e232 0x62c15e 0x1c62c8a 0x62a53d 0x4c6a41
#	0x62c15d	testing.(*T).Run+0x9fd								/usr/local/go/src/testing/testing.go:2005
#	0x1c62c89	github.com/prometheus/prometheus/storage/remote.TestReshardPartialBatch+0xa9	/__w/prometheus/prometheus/storage/remote/queue_manager_test.go:627
#	0x62a53c	testing.tRunner+0x21c								/usr/local/go/src/testing/testing.go:1934

1 @ 0x4bdf2e 0x44e685 0x44e232 0x62c765 0x629e53 0x1c648bf 0x62a53d 0x4c6a41
#	0x62c764	testing.(*testState).waitParallel+0x124						/usr/local/go/src/testing/testing.go:2116
#	0x629e52	testing.(*T).Parallel+0x532							/usr/local/go/src/testing/testing.go:1709
#	0x1c648be	github.com/prometheus/prometheus/storage/remote.TestDisableReshardOnRetry+0x3e	/__w/prometheus/prometheus/storage/remote/queue_manager_test.go:780
#	0x62a53c	testing.tRunner+0x21c								/usr/local/go/src/testing/testing.go:1934

1 @ 0x4bdf2e 0x44e685 0x44e232 0x633eb0 0x1a76e65 0x4c6a41
#	0x633eaf	testing/synctest.testingSynctestTest+0x60f	/usr/local/go/src/testing/testing.go:2047
#	0x1a76e64	testing/synctest.Test.func1+0x44		/usr/local/go/src/testing/synctest/synctest.go:283

1 @ 0x4bdf2e 0x4992f7 0x16a8c5b 0x4c6a41
#	0x16a8c5a	github.com/prometheus/prometheus/util/logging.(*Deduper).run+0x19a	/__w/prometheus/prometheus/util/logging/dedupe.go:122

1 @ 0x4bdf2e 0x4992f7 0x197d68c 0x4c6a41
#	0x197d68b	github.com/open-telemetry/opentelemetry-collector-contrib/processor/deltatocumulativeprocessor.(*deltaToCumulativeProcessor).Start.func1+0x26b	/go/pkg/mod/github.com/open-telemetry/opentelemetry-collector-contrib/processor/deltatocumulativeprocessor@v0.145.0/processor.go:194

1 @ 0x4bdf2e 0x4992f7 0x1c12e57 0x1c11f6c 0x4c6a41
#	0x1c12e56	github.com/prometheus/prometheus/storage/remote.(*shards).stop+0x356			/__w/prometheus/prometheus/storage/remote/queue_manager.go:1329
#	0x1c11f6b	github.com/prometheus/prometheus/storage/remote.(*QueueManager).reshardLoop+0xeb	/__w/prometheus/prometheus/storage/remote/queue_manager.go:1236

1 @ 0x4bdf2e 0x49a6bd 0x49a694 0x4bf865 0x4d6090 0x4d5db5 0x4d6c69 0x1c62b7f 0x4c6a41
#	0x4bf864	internal/sync.runtime_SemacquireMutex+0x24						/usr/local/go/src/runtime/sema.go:95
#	0x4d608f	internal/sync.(*Mutex).lockSlow+0x20f							/usr/local/go/src/internal/sync/mutex.go:149
#	0x4d5db4	internal/sync.(*Mutex).Lock+0x54							/usr/local/go/src/internal/sync/mutex.go:70
#	0x4d6c68	sync.(*Mutex).Lock+0x28									/usr/local/go/src/sync/mutex.go:46
#	0x1c62b7e	github.com/prometheus/prometheus/storage/remote.TestReshardRaceWithStop.func1.1+0x23e	/__w/prometheus/prometheus/storage/remote/queue_manager_test.go:603

1 @ 0x4bdf2e 0x4c2506 0x1a76dc5 0x1c606a6 0x1c60634 0x62a53d 0x4c6a41
#	0x4c2505	internal/synctest.Run+0x1e5						/usr/local/go/src/runtime/synctest.go:218
#	0x1a76dc4	testing/synctest.Test+0x124						/usr/local/go/src/testing/synctest/synctest.go:282
#	0x1c606a5	github.com/prometheus/prometheus/util/testutil/synctest.Test+0xa5	/__w/prometheus/prometheus/util/testutil/synctest/synctest.go:24
#	0x1c60633	github.com/prometheus/prometheus/storage/remote.TestShutdown.func1+0x33	/__w/prometheus/prometheus/storage/remote/queue_manager_test.go:476
#	0x62a53c	testing.tRunner+0x21c							/usr/local/go/src/testing/testing.go:1934

1 @ 0x4bdf2e 0x4c2718 0x1c609ac 0x1c609a7 0x1c609a6 0x62a53d 0x4c6a41
#	0x4c2717	internal/synctest.Wait+0x77							/usr/local/go/src/runtime/synctest.go:298
#	0x1c609ab	testing/synctest.Wait+0x2cb							/usr/local/go/src/testing/synctest/synctest.go:302
#	0x1c609a6	github.com/prometheus/prometheus/util/testutil/synctest.Wait+0x2c6		/__w/prometheus/prometheus/util/testutil/synctest/synctest.go:28
#	0x1c609a5	github.com/prometheus/prometheus/storage/remote.TestShutdown.func1.1+0x2c5	/__w/prometheus/prometheus/storage/remote/queue_manager_test.go:493
#	0x62a53c	testing.tRunner+0x21c								/usr/local/go/src/testing/testing.go:1934

--- FAIL: TestReshardPartialBatch (13.16s)
    --- FAIL: TestReshardPartialBatch/prometheus.WriteRequest (6.72s)
        queue_manager_test.go:656: Deadlock between sending and stopping detected
FAIL
```

--- a/storage/remote/queue_manager_test.go
+++ b/storage/remote/queue_manager_test.go
@@ -604,7 +604,6 @@ func TestReshardRaceWithStop(t *testing.T) {
 }
 
 func TestReshardPartialBatch(t *testing.T) {
-	t.Parallel()
 	for _, protoMsg := range []remoteapi.WriteMessageType{remoteapi.WriteV1MessageType, remoteapi.WriteV2MessageType} {
 		t.Run(fmt.Sprint(protoMsg), func(t *testing.T) {
 			recs := testwal.GenerateRecords(recCase{

